### ISSUE TITLE
Conda Build doesn't allow restricting exact build for a build dependency.

### ISSUE THREAD

**Issue by nehaljwani** [conda/conda-build#3233]
### Actual Behavior

Conda build ignores the build string in the build/host dep.

### Expected Behavior

Conda build should honor the exact build string.

### Steps to Reproduce

Create a recipe with a build/host dependency which is pinned upto the build string, try to build it.

##### `Output of conda info`

```
[root@3890f5de353f recipe]# cat meta.yaml 
package:
  name: test_strict_build
  version: 1.0.0

requirements:
  build:
    - zlib ==1.2.8_0

[root@3890f5de353f recipe]# conda build .                                                                                                                                                                          
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Attempting to finalize metadata for test_strict_build
INFO:conda_build.metadata:Attempting to finalize metadata for test_strict_build
Solving environment: ...working... done
Solving environment: ...working... done
BUILD START: ['test_strict_build-1.0.0-0.tar.bz2']
Solving environment: ...working... done

## Package Plan ##

  environment location: /opt/conda/conda-bld/test_strict_build_1540445804839/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho
ld_placehold_placehold_placehold_placehold_placehold_placehold_place


The following NEW packages will be INSTALLED:

    zlib: 1.2.8-3 defaults  <-------------- I asked for build 0!
```

**Comment by msarahan** [comment]
Wrong syntax. Should be

- zlib 1.2.8 0

I think we have issues filled here about accepting other conda specs or maybe even pip format. I'm afraid that might be a large pile of work to fix lots of subtle hardcoded assumptions about .split() behavior.

**Comment by msarahan** [comment]
Upon further though, I'm confused as to why you didn't get some sort of UnsatisfiableError if conda/conda-build didn't understand your spec.  This is definitely worth looking into.

**Comment by bkreider** [comment]
I think this issue is closed by PR #3236 (#3234 was pulled into that one i think).
